modules/sovrnBidAdapter.js

```diff
@@ -1,156 +1,82 @@
-var CONSTANTS = require('src/constants.json');
-var utils = require('src/utils.js');
-var bidfactory = require('src/bidfactory.js');
-var bidmanager = require('src/bidmanager.js');
-var adloader = require('src/adloader');
-var adaptermanager = require('src/adaptermanager');
-
-/**
- * Adapter for requesting bids from Sovrn
- */
-var SovrnAdapter = function SovrnAdapter() {
-  var sovrnUrl = 'ap.lijit.com/rtb/bid';
-
-  function _callBids(params) {
-    var sovrnBids = params.bids || [];
-
-    _requestBids(sovrnBids);
-  }
-
-  function _requestBids(bidReqs) {
-    // build bid request object
-    var domain = window.location.host;
-    var page = window.location.pathname + location.search + location.hash;
-
-    var sovrnImps = [];
-
-    // build impression array for sovrn
+import * as utils from 'src/utils';
+import { registerBidder } from 'src/adapters/bidderFactory';
+import { BANNER } from 'src/mediaTypes';
+import { REPO_AND_VERSION } from 'src/constants';
+
+export const spec = {
+  code: 'sovrn',
+  supportedMediaTypes: [BANNER],
+
+  /**
+   * Check if the bid is a valid zone ID in either number or string form
+   * @param {object} bid the Sovrn bid to validate
+   * @return boolean for whether or not a bid is valid
+   */
+  isBidRequestValid: function(bid) {
+    return !!(bid.params.tagid && !isNaN(parseFloat(bid.params.tagid)) && isFinite(bid.params.tagid));
+  },
+
+  /**
+   * Format the bid request object for our endpoint
+   * @param {BidRequest[]} bidRequests Array of Sovrn bidders
+   * @return object of parameters for Prebid AJAX request
+   */
+  buildRequests: function(bidReqs) {
+    let sovrnImps = [];
     utils._each(bidReqs, function (bid) {
-      var tagId = utils.getBidIdParameter('tagid', bid.params);
-      var bidFloor = utils.getBidIdParameter('bidfloor', bid.params);
-      var adW = 0;
-      var adH = 0;
-
-      // sovrn supports only one size per tagid, so we just take the first size if there are more
-      // if we are a 2 item array of 2 numbers, we must be a SingleSize array
-      var bidSizes = Array.isArray(bid.params.sizes) ? bid.params.sizes : bid.sizes;
-      var sizeArrayLength = bidSizes.length;
-      if (sizeArrayLength === 2 && typeof bidSizes[0] === 'number' && typeof bidSizes[1] === 'number') {
-        adW = bidSizes[0];
-        adH = bidSizes[1];
-      } else {
-        adW = bidSizes[0][0];
-        adH = bidSizes[0][1];
-      }
-
-      var imp =
-        {
-          id: bid.bidId,
-          banner: {
-            w: adW,
-            h: adH
-          },
-          tagid: tagId,
-          bidfloor: bidFloor
-        };
-      sovrnImps.push(imp);
+      sovrnImps.push({
+        id: bid.bidId,
+        banner: { w: 1, h: 1 },
```

Comment:
@tedrand Is this hardcoded as 1x1 because Sovrn knows the size based on the ID?  It seems like the bid.sizes and bid.params.sizes properties will be totally ignored by the adapter.  Is that correct?

@mkendall07 If that’s the case, then https://github.com/prebid/Prebid.js/blob/master/modules/sovrnBidAdapter.md should be updated to omit the sizes fields, right?